libraries/azure_vpn_gateways.rb

```diff
@@ -0,0 +1,46 @@
+require 'azure_generic_resources'
+
+class AzureVPNGateways < AzureGenericResources
+  name 'azure_vpn_gateways'
+  desc 'Verifies settings for AKS VPN Gateways'
+  example <<-EXAMPLE
+    azure_vpn_gateways(resource_group: 'example') do
+      it{ should exist }
+    end
+  EXAMPLE
+
+  attr_reader :table
+
+  def initialize(opts = {})
+    # Options should be Hash type. Otherwise Ruby will raise an error when we try to access the keys.
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+
+    opts[:resource_provider] = specific_resource_constraint('Microsoft.Network/virtualNetworkGateways', opts)
+
```

Comment:
`resource_group` should be mandatory. See, [resource creation guideline](https://github.com/inspec/inspec-azure/blob/master/docs/resource_creation_guide.md#required-parameters) and an example usage [here](https://github.com/inspec/inspec-azure/blob/master/libraries/azure_mysql_databases.rb#L20).

Reply:
It seems to work as an optional parameter when I've been testing it and behaving as expected.